packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -177,106 +176,47 @@ export async function handleRoute({
 	const middleware = (await loadMiddleware(loader)).onRequest;
 	const locals = Reflect.get(incomingRequest, clientLocalsSymbol);
 
-	if (!matchedRoute) {
```

Comment:
`matchedRoute` is always truthy because of `ensure404Route`, so this branch is never hit